web-common/src/features/dashboards/stores/get-explore-state-from-yaml-config.ts

```diff
@@ -1,24 +1,179 @@
-import { convertPresetToExploreState } from "@rilldata/web-common/features/dashboards/url-state/convertPresetToExploreState";
-import { getDefaultExplorePreset } from "@rilldata/web-common/features/dashboards/url-state/getDefaultExplorePreset";
-import type {
-  V1ExploreSpec,
-  V1MetricsViewSpec,
-  V1TimeRangeSummary,
+import { SortDirection } from "@rilldata/web-common/features/dashboards/proto-state/derived-types";
+import { getGrainForRange } from "@rilldata/web-common/features/dashboards/stores/get-rill-default-explore-state";
+import type { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
+import { getValidComparisonOption } from "@rilldata/web-common/features/dashboards/time-controls/time-range-store";
+import { ToLegacySortTypeMap } from "@rilldata/web-common/features/dashboards/url-state/legacyMappers";
+import { FromURLParamTimeGrainMap } from "@rilldata/web-common/features/dashboards/url-state/mappers";
+import { arrayUnorderedEquals } from "@rilldata/web-common/lib/arrayUtils";
+import { ISODurationToTimePreset } from "@rilldata/web-common/lib/time/ranges";
+import { isoDurationToFullTimeRange } from "@rilldata/web-common/lib/time/ranges/iso-ranges";
+import {
+  type DashboardTimeControls,
+  TimeRangePreset,
+} from "@rilldata/web-common/lib/time/types";
+import { DashboardState_ActivePage } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import {
+  V1ExploreComparisonMode,
+  type V1ExploreSpec,
+  type V1TimeRangeSummary,
 } from "@rilldata/web-common/runtime-client";
 
 export function getExploreStateFromYAMLConfig(
-  metricsViewSpec: V1MetricsViewSpec,
   exploreSpec: V1ExploreSpec,
   timeRangeSummary: V1TimeRangeSummary | undefined,
```

Comment:
I'm wondering if we need the `timeRangeSummary` parameter here? Since the Explore YAML `partialExploreState` is ultimately merged with `rillDefaultState`, then we'd apply our opinionated time range logic (which leverages the `timeRangeSummary`) in the `rillDefaultState` section.

If this is scope expansion and should be deferred to another PR, that's fine.

Reply:
We do a dumb merge right now. But we need to set certain other fields based on time range selected here. We need timeRangeSummary do those calculations. This is something that could be moved to post merge "validation". We could do a follow up once that is in.